### PR TITLE
feat: allow types workflow to pick up preleases

### PR DESCRIPTION
**Pull request by kylelaker**

### CODE REVIEW

File: .github/workflows/update-types.yml [hunk 1]
```diff
@@ -6,25 +6,54 @@ on:
...
+            let url = undefined;
+            for (const asset of release.assets) {
+              if (asset.name == `oscal-${version}.zip`) {
+                url = asset.browser_download_url;
+                break;
+              }
+            }
+            if (!url) {
+              console.log("WTF?!");
+            }
```
Comment (kylelaker) [inline .github/workflows/update-types.yml:49]:
```suggestion
            if (!url) {
              console.log(`Failed to find a matching download URL for ${version}`);
            }
```

**Review by Bronstrom — COMMENTED** [review]
Overall, this looks great! I would remove these `console.log`s though.

File: .github/workflows/update-types.yml [hunk 2]
```diff
@@ -6,25 +6,54 @@ on:
...
+        required: true
+        default: false
+        type: boolean
 
 permissions:
   contents: read
 
+env:
+  OSCAL_REPO: usnistgov/OSCAL
+
```
Comment (kylelaker) [inline .github/workflows/update-types.yml:21]:
```suggestion
```

File: .github/workflows/update-types.yml [hunk 3]
```diff
@@ -15,16 +21,36 @@ jobs:
...
+        id: fetch-data
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const prerelease = context.payload?.inputs?.prerelease ?? false;
+            const requestParams = { owner: "usnistgov", repo: "OSCAL" };
+            const release = prerelease
+              ? (await github.rest.repos.listReleases(requestParams)).data[0]
+              : (await github.rest.repos.getLatestRelease(requestParams)).data;
+            console.log(JSON.stringify(release));
```
Comment (kylelaker) [inline .github/workflows/update-types.yml:34]:
```suggestion
```

**Review by Bronstrom — APPROVED** [review]

**Review by tuckerzp — APPROVED** [review]

**Pull request merged**